lib/subproviders/geth_api_double.js

```diff
@@ -227,9 +229,19 @@ GethApiDouble.prototype.eth_getTransactionByHash = function(hash, callback) {
     var result = null;
 
     if (receipt) {
+      // if there is no block then the transaction is still pending
+      if (!receipt.block) {
+        // assemble the block object with null values for pending transactions
+        receipt.block = {
```

Comment:
Would just using `new Block()` here work instead of mocking a `Block` object?

Reply:
I don't think that using `new Block()` here would work because it generates a `hash` for you and I can't seem to overwrite it by passing in null. I am able to overwrite `header.number` and set that to `null` but not able to with `hash` and that needs to be null when it's pending.